UniAtHome/UniAtHome.BLL/Services/UniversityCreationService.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UniAtHome.BLL.DTOs.UniversityCreation;
+using UniAtHome.BLL.Exceptions;
+using UniAtHome.BLL.Interfaces;
+using UniAtHome.DAL.Entities;
+using UniAtHome.DAL.Interfaces;
+
+namespace UniAtHome.BLL.Services
+{
+    public sealed class UniversityCreationService : IUniversityCreationService
+    {
+        private readonly IRepository<UniversityCreateRequest> requestsRepository;
+
+        public UniversityCreationService(IRepository<UniversityCreateRequest> requestsRepository)
+        {
+            this.requestsRepository = requestsRepository;
+        }
+
+        public async Task AddRequestAsync(UniversityCreateDTO creationInfo)
+        {
+            var createRequest = new UniversityCreateRequest
+            {
+                UniversityName = creationInfo.UniversityName,
+                SubmitterFirstName = creationInfo.SubmitterFirstName,
+                SubmitterLastName = creationInfo.SubmitterLastName,
+                Email = creationInfo.Email,
+                Comment = creationInfo.Comment,
+                Submitted = DateTime.Now
+            };
```

Comment:
👎 what if in the future we would have to map this type of dto object to UniversityCreateRequest object? if u don't like using automapper, it's better to at least extract such functionality in separate function, so that there will be no need to change many blocks of code. And I prefer unification: if we use automapper, we should continue using it, if there's no urgent need to act differently (I don't see such need here)

Reply:
Automapper sucks especially when the entities are too small as we have now. Adding new entites requires additional time wasted on automapper and redundant references. And when you try to do refactoring, you waste time to delete automapper's configuration and all the references to the entity. I suggest adding entity to the automapper when it exists more than two weeks (100% won't be deleted), we're using it in multiple places or it has too many fields. I hate spaghetti code created by making redundant dependencies all over the project and I WILL make short and plain mappings as far as they don't repeat.